inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -938,3 +938,43 @@ def normalize_collaborations(obj, eng):
         collaboration['value'] = collaboration_normalized_name
         collaboration['record'] = response[0].self.to_dict()
     return obj
+
+
+def core_selection_wf_already_created(control_number):
+    query_wf_with_cn = Q("match", metadata__control_number=control_number)
+    query_core_selection_wf = Q("match", _workflow__workflow_name="CORE_SELECTION")
+    query = query_wf_with_cn & query_core_selection_wf
+
+    return Search(index="holdingpen-hep", using=current_search_client).query(query).source(False).execute()
+
+
+def create_core_selection_wf(obj, eng):
+    record_control_number = obj.data.get('control_number')
+    if not record_control_number:
+        raise MissingRecordControlNumber
+    if obj.data.get('core') or core_selection_wf_already_created(record_control_number):
+        LOGGER.info("No core selection needed for %s workflow with record %s",)
+        return obj
+
+    extra_data_to_clean = ['_task_history', '_error_msg', 'source_data', 'restart-count']
+    extra_data = dict(obj.extra_data)
```

Comment:
this only makes a copy if `extra_data` is not a `dict` itself. Otherwise, you need to do `obj.extra_data.copy()`